app/models/miq_server/worker_management/systemd.rb

```diff
@@ -3,6 +3,19 @@ def sync_from_system
     self.miq_services = systemd_services.select { |unit| manageiq_service?(unit) }
   end
 
+  def sync_starting_workers!(starting)
+    sync_from_system
+    starting.each do |worker|
+      next if worker.class.rails_worker?
+
+      systemd_worker = miq_services.find { |w| w[:name].include?(worker[:guid]) }
```

Comment:
It would probably be helpful to build a hash of the systemd units by the worker guid so you could just do `miq_services_by_guid[worker[:guid]]`

Something like
`miq_services_by_guid = miq_services.index_by { |w| w[:name].match(/^.+@(?<guid>.+)\.service$/).named_captures["guid"] }`